test/e2e/pvc_test.go

```diff
@@ -0,0 +1,255 @@
+package e2e_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	"github.com/openshift/cluster-image-registry-operator/test/framework"
+)
+
+func testDefer(t *testing.T, client *framework.Clientset) {
+	if t.Failed() {
+		scList, err := client.StorageClasses().List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the storage classes: %s", err)
+		} else {
+			framework.DumpYAML(t, "storageclasses", scList)
+		}
+
+		pvList, err := client.PersistentVolumes().List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the persistent volumes: %s", err)
+		} else {
+			framework.DumpYAML(t, "persistentvolumes", pvList)
+		}
+
+		pvcList, err := client.PersistentVolumeClaims(imageregistryv1.ImageRegistryOperatorNamespace).List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the persistent volume claims: %s", err)
+		} else {
+			framework.DumpYAML(t, "persistentvolumeclaims", pvcList)
+		}
+	}
+	framework.MustRemoveImageRegistry(t, client)
+}
+
+func createPV(t *testing.T, storageClass string) error {
+	client := framework.MustNewClientset(t, nil)
+	name := ""
+
+	for i := 0; i < 100; i++ {
+		pvName := fmt.Sprintf("pv-%s", rand.String(64))
```

Comment:
including `i` in the name would further ensure uniqueness

Reply:
@bparees I use 64 random bits with 100 retries. If you think that adding one number will greatly increase randomization, ok I will add it.

Reply:
it was only relevant when i thought you were creating 100 at once.  since you're only trying to create one, it's less important (and yes, it wasn't that important in the first place, but given that you had the entropy available, it seemed to make sense to use it).

Reply:
Why not use uuid like we do in s3? (you can truncate it to the specified length)

Reply:
@coreydaley I think the kubernetes module is more useful here. It was created specifically for such cases and I'm sure everything is done right there.